### PR TITLE
Update release instructions with more details.

### PR DESCRIPTION
**Pull request by bdferris-v2**

### CODE REVIEW

**Comment by bdferris-v2** [comment]
@jameslinjl can you take a quick look at this documentation update?

File: golang/UPDATING.md [hunk 1]
```diff
@@ -18,4 +18,27 @@
 
 ## Publishing a new release
 
-Ask someone with write permission for the repo to follow the [instructions for publishing Go modules](https://go.dev/blog/publishing-go-modules).
\ No newline at end of file
+Publishing can only be performed by someone with commit privileges to the repo.
+
+1. Ensure you are the in branch (likely `master`) and at the commit you'd like to tag as a release version.
```
Comment (jameslinjl) [inline golang/UPDATING.md:23]: typo?

**Review by jameslinjl — APPROVED** [review]
lgtm overall! just the first sentence reads a little funny / may have a typo

File: golang/UPDATING.md (hunk 1 above)
Comment (bdferris-v2) [reply golang/UPDATING.md]: Ha I tried to fix this typo once already and got the order wrong :)

**Comment by bdferris-v2** [comment]
@isabelle-dr this doc update has an LGTM from James but needs an official review from someone with commit privileges (and I can't review my own PR).  Any chance you could sign off?

**Review by isabelle-dr — APPROVED** [review]
LGTM, merging this PR. Thanks for updating the instructions @bdferris-v2 !

**Pull request merged**